### ISSUE TITLE
Names of mount points instead of device names.

### ISSUE THREAD

**Issue by otherguy** [victorgama/howe#4]
Currently, devices names are used in the disks widget. It would be great if we could (maybe optionally with a `mountpoints: yes` flag) see the mountpoints instead of the -- sometimes not so descriptive -- device names.

```
Filesystems                      Size  Used  Free  Use%
  /dev/sda1                      497M  218M  279M   44%
  [===================================================]
  /dev/mapper/rootvg-rootlv      7.0G  3.0G  4.0G   43%
  [===================================================]
  /dev/mapper/rootvg-homelv      2.0G   34M  2.0G    2%
  [===================================================]
  /dev/mapper/rootvg-tmplv       1.0G   33M  1.0G    4%
  [===================================================]
  /dev/mapper/rootvg-varlv       4.0G  1.1G  2.9G   27%
  [===================================================]
```

It's fairly simple to find the mount point for a device:

`$ findmnt --first-only --noheadings --notruncate  /dev/mapper/rootvg-varlv  | awk '{print $1}'`

If nothing is returned, the device might not be mounted, so just print the device name.

In my case that would result in the much more readable:

```
Filesystems                      Size  Used  Free  Use%
  /boot      497M  218M  279M   44%
  [===================================================]
  /          7.0G  3.0G  4.0G   43%
  [===================================================]
  /home      2.0G   34M  2.0G    2%
  [===================================================]
  /tmp       1.0G   33M  1.0G    4%
  [===================================================]
  /var       4.0G  1.1G  2.9G   27%
  [===================================================]
```

**Comment by victorgama** [comment]
@otherguy Sorry taking so long to address this. I liked the idea!

Question: I'm thinking about assuming any path lacking a `/dev` prefix as a mountpoint. Do you think that's safe?
Also, I'm inclined to parse `/proc/mounts` over relying on `findmnt`. Would that work in your case?

**Comment by otherguy** [comment]
Don't worry @victorgama -- it was Christmas after all 🎄🙂 

Sure, parsing `/proc/mounts´ is fine, but I believe the `mount` package *always* comes with `findmnt`, giving you an easier to use and consistent interface.

**Comment by victorgama** [comment]
@otherguy Hey there! Sorry for the delay, I was overwhelmed at work.
Would you mind sharing the content of your `/proc/mounts`?
In the meantime I'll run a few tests to compare how long would it take to parse `/proc/mounts` over calling `findmnt`. Wouldn't like to add extra milliseconds between connecting to a server and being able to use the shell. 🤔

**Comment by otherguy** [comment]
@victorgama no problem, I was too ;)

Here's an example:


```
$ cat /proc/mounts
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
proc /proc proc rw,relatime 0 0
udev /dev devtmpfs rw,nosuid,relatime,size=16383932k,nr_inodes=4095983,mode=755 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,nosuid,noexec,relatime,size=3280624k,mode=755 0 0
/dev/md2 / ext4 rw,relatime,data=ordered 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,nosuid,nodev 0 0
tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime,size=5120k 0 0
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb,release_agent=/run/cgmanager/agents/cgm-release-agent.hugetlb 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset,clone_children 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event,release_agent=/run/cgmanager/agents/cgm-release-agent.perf_event 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids,release_agent=/run/cgmanager/agents/cgm-release-agent.pids 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=32,pgrp=1,timeout=0,minproto=5,maxproto=5,direct 0 0
mqueue /dev/mqueue mqueue rw,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,relatime 0 0
tracefs /sys/kernel/debug/tracing tracefs rw,relatime 0 0
/dev/md1 /boot ext3 rw,relatime,data=ordered 0 0
cgmfs /run/cgmanager/fs tmpfs rw,relatime,size=100k,mode=755 0 0
none /var/lib/docker/aufs/mnt/48cf62afe205c948b63918a00d3aff19214bd8d6d7af3186f2bd290bcbc519c9 aufs rw,relatime,si=29428cccb5698340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/5c64a70a769e1c22fcaa428673acedff9b5112916201c732bbc4dcaeb97f2ea8 aufs rw,relatime,si=29428ccc17078340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/c5848d27cdfc96bea46df946e554c24df573a8a628de7ea2a0d0956a551000f5 aufs rw,relatime,si=29428cccb569a340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/2d97f8da5a7e9595d038fe7315ad2cac2a08c90e90188a8a3de91ffbe79e14f5 aufs rw,relatime,si=29428ccf38d05340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/b32c68bf5601939b9a3a3a3b76cda52ee0ddd00c25b760be4bbbef9159469f3d aufs rw,relatime,si=29428ccd5dec5340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/c1cb512e881b33aeddd1f82eee3e07d9f78c85544ab0ae661202e2c33756c37a aufs rw,relatime,si=29428ccd5dec1340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/586b98627c77b4cc8684c0816108fdc6be1ae0cd8836cc7d550ab22df8e60b17 aufs rw,relatime,si=29428cc9ad022340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/710a9b8376aa8b410da35455df3f974236d911476b50544662683aff8e6f9f53 aufs rw,relatime,si=29428ccee0ff0340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/4aadb091939f80318e098fa36f598d90e903165f9a5b5c370ed1a1549a41674b aufs rw,relatime,si=29428ccc1700f340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/892ff46ab364a2c30c62ac64c85336afccbddddec952f9bbc28d539b4c9db967 aufs rw,relatime,si=29428ccc17009340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/45192eae2f13f331bfc5c61fa242e69fb7852eab5cb577ddc9e0ad9811fee779 aufs rw,relatime,si=29428cce8ab9e340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/998011f592cbda9c32e1fba9feb8f955a43f6539e8a89f3ed83bce9f4246a0c2 aufs rw,relatime,si=29428cce8ab9a340,dio,dirperm1 0 0
none /var/lib/docker/aufs/mnt/d5c4b6d03faf1ef85a0153b09b08220a78bcb618513f4ed26b33d5ccbe18bd67 aufs rw,relatime,si=29428cce8ab9c340,dio,dirperm1 0 0
shm /var/lib/docker/containers/440054e41dc3f84d8aef2b6aadd28233155c69d324c57503d74f90b8a981eb05/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
shm /var/lib/docker/containers/b1fe8b73e04a021d69d3b1b79e33ed19be2df0c4ad93e77700691a3461e7485b/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/224a1f5dc06a nsfs rw 0 0
nsfs /run/docker/netns/920956113d66 nsfs rw 0 0
shm /var/lib/docker/containers/0ddc81f03a0d4429f0eae3206b2d20693a4a1f35a332be0ad79ab8ad5b98cf3e/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
shm /var/lib/docker/containers/644873504fdba9331840f835bc198d1ca567538aa6e0015abf1b1595a87a5471/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/1e374ae893e3 nsfs rw 0 0
nsfs /run/docker/netns/c0a777fcf70d nsfs rw 0 0
shm /var/lib/docker/containers/5439c7a24f07b96365e675853c2990062771152f963dd477f2d133c2450b132b/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
shm /var/lib/docker/containers/3194df531a73584cbd00c950f855463dd37ca1aade548d712cfdeff1a400714a/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
shm /var/lib/docker/containers/8959239b515cd94b1a81ed9b9712445e99682fa51132a62fbab8415f527ef550/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/995bf304995c nsfs rw 0 0
nsfs /run/docker/netns/4b4e8b4dd78f nsfs rw 0 0
nsfs /run/docker/netns/33ba6411ee56 nsfs rw 0 0
shm /var/lib/docker/containers/5b8080a252738d56e69f53e0886d6c5b75a149e557e60f7a4c0177a228642724/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/37adb01bb427 nsfs rw 0 0
shm /var/lib/docker/containers/5fdccfba075c9f5f287d7806c310f941a8f7d31ffad86250a1feed3f6c9f793e/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/e59811d19257 nsfs rw 0 0
shm /var/lib/docker/containers/514f0193c207597a36f52c063722a73663bd92e86b9e437c49a627c22d73c661/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/23cb6238c281 nsfs rw 0 0
shm /var/lib/docker/containers/4d9acb188667a753e3a87ddefe3b9c62ef7e79177550c06e9ad4a224c35e5461/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
shm /var/lib/docker/containers/f4ae70b144be0d4e68e49102c0c8be097c0a82907cfa6b3069ec35fad13c2817/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/abe97abd9d4c nsfs rw 0 0
nsfs /run/docker/netns/ff7a993f82d5 nsfs rw 0 0
none /var/lib/docker/aufs/mnt/c3f39565232f943995814de53b72b8ef53f11a5e186d8b0a3a86d7acb47b5b12 aufs rw,relatime,si=29428cc9adbed340,dio,dirperm1 0 0
shm /var/lib/docker/containers/0a2d80ff5667cb31a731617028e87e3ad7d96aa3e40b4852fa952b6dcb337852/mounts/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
nsfs /run/docker/netns/bd1a7b4c204a nsfs rw 0 0
tmpfs /run/user/0 tmpfs rw,nosuid,nodev,relatime,size=3280624k,mode=700 0 0
```

**Comment by victorgama** [comment]
@otherguy It seems that one does not have `/dev/mapper` entries. Mind sharing one which does? 😬

**Comment by otherguy** [comment]
@victorgama I can but `/dev/mapper` entries are just LVM logical volumes. The feature should be universal and not only work on LVM mounts.

The columns in `/proc/mounts` are as follows (separated by 1 or more whitespace characters `\s+` -- tabs and spaces both are allowed):

- The 1st column specifies the device that is mounted.
- The 2nd column reveals the mount point.
- The 3rd column tells the file-system type.
- The 4th column tells you if it is mounted read-only (`ro`) or read-write (`rw`) along with other special flags that are comma separated. E.g.: `nosuid,nodev,noexec,relatime`
- The 5th and 6th columns are dummy values designed to match the format used in `/etc/mtab`

So right now you are displaying the `device`, gathered from _sigar_ as I noticed. More helpful is the mountpoint though.

Not sure how _sigar_ handles this but I assume it ignores "special" devices. If the first column is not a pointer on a filesystem, it's a "special" device, like all the Docker entries that you see in my example.

So in the list above, I removed all lines that do not begin with a slash (`^[^/].*$`) which leaves us with:

```
/dev/md2 / ext4 rw,relatime,data=ordered 0 0
/dev/md1 /boot ext3 rw,relatime,data=ordered 0 0
```

That means:
- The _device_ `/dev/md2` is mounted on `/` and it's an `ext4` volume that can be written (`rw`)
- The _device_ `/dev/md1` is mounted on `/boot` and it's an `ext3` volume that can be written (`rw`)

My preferred output would now be:

```
Mounts                         Size  Used  Free  Use%

/boot                          497M  218M  279M   44%
[===================================================]

/ (root)                       7.0G  3.0G  4.0G   43%
[===================================================]
```

Maybe make it configurable to show the device or mountpoint (like I mentioned initially, with `mountpoints: yes`), that should make everyone happy.

**Comment by victorgama** [comment]
@otherguy Implemented and [available to download](https://github.com/victorgama/howe/releases/tag/v0.2.1). 🙌 

No flag needed, just use your mountpoint as the disk name. It will first attempt to find a mountpoint, and falls back to the previous behaviour if none is found:

```yaml
- type: disks
  disks:
#  - /dev/vda1
   - /
```

Again, thank you so much for the feedback!

**Comment by otherguy** [comment]
@victorgama thanks! But this won't work if I don't specify any disks (instead displaying all of them), correct?

**Comment by victorgama** [comment]
@otherguy That's unfortunately correct.
But it may still have a workaround: we can check whether both device and mountpoint begins with a forward slash when attempting to list all disks, but since we're relying on `/proc/mounts` to do that, I think that the "cleaner" version will only work on Linux for now.

I'll give it a try later today, and, if results are positive, I'll push another version right after. In the meantime, I'll leave this issue open.

**Comment by heyvito** [comment]
Fixed in [v0.2.2](https://github.com/victorgama/howe/releases/tag/v0.2.2)
Now `'*'` will list all known mountpoints instead of known devices.
(Which may cause devices to be listed twice, but I'm not sure what to do in this case)

**Comment by ghost** [comment]
Pretty awesome :)

```
Filesystems    Size  Used  Free  Use%
  /            7.0G  3.0G  4.0G   43%
  [=================================]
  /var         4.0G  1.1G  2.9G   28%
  [=================================]
  /home        2.0G   59M  1.9G    3%
  [=================================]
  /tmp         1.0G   34M  1.0G    4%
  [=================================]
  /boot        497M  219M  278M   45%
  [=================================]
  /perf        497M  303M  194M   62%
  [=================================]
  /tools         0     0     0     0%
  [=================================]
```
